query.go

```diff
@@ -662,6 +670,34 @@ checkResponses:
 	}
 }
 
+// getFilterFromCache returns a filter from ChainService's FilterCache if it
+// exists, returning nil and error if it doesn't.
+func (s *ChainService) getFilterFromCache(blockHash *chainhash.Hash,
+	filterType filterdb.FilterType) (*gcs.Filter, error) {
+
+	cacheKey := filterCacheKey{blockHash: blockHash, filterType: filterType}
+
+	filterValue, err := s.FilterCache.Get(cacheKey)
+	if err != nil {
+		return nil, err
+	}
+
+	cachedFilter, ok := filterValue.(*cache.CacheableFilter)
```

Comment:
Just return `filterValue.(*cache.CacheableFilter).Filter` at this point. If the cache contains something that is not a filter we should panic.